### PR TITLE
Ignore client closing error in `tbot` CA Watcher when certificates renew

### LINKED ISSUE #14753: `tbot` CA rotation watcher emits client connection error after renewal
```
Error occured whilst watching CA rotations, retrying in 1.294469236s. error:[
ERROR REPORT: 
Original Error: *status.Error rpc error: code = Canceled desc = grpc: the client connection is closing
Stack Trace:
/go/src/github.com/gravitational/teleport/api/client/streanwatcher.go:87 [github.com/gravitational/teleport/api/client.(*streamWatcher).receiveEvents](http://github.com/gravitational/teleport/api/client.(*streamWatcher).receiveEvents)
/opt/go/src/runtime/asm_amd64.s:1571 runtime.goexit
User Message: rpc error: code = Canceled desc = grpc: the client connection is closing] tbot/ca_rotation.go:117
```

Is emitted after each renewal. This is because when we renew, we replace the central client in `tbot` with one using the new certificates, and as part of this, we call `client.Close()` on the old client. This causes the open connection used by the CA rotation watcher to close, throwing this error. Whilst this error looks concerning, it's mostly innocent. The connection re-opens itself, although, there is a small risk of this coinciding with a rotation.

I see two options:
- Smarter closing, that doesn't close it if there is an open watcher using that client.
- Swallow client connection error and reconnect immediately rather than retrying after some period, as we know what causes this and it is a special case.

Whilst the first option seems the "cleanest", I'm concerned it might actually be fiddly to keep track of if a watcher is using the client, and that we risk leaking clients if we don't detect the watcher going away. The second option is easier, and does always ensure that the watcher is using the latest credentials (even though this doesn't really make much of a difference).

### PR DESCRIPTION
**Pull request by strideynet**
Closes https://github.com/gravitational/teleport/issues/14753

This error is caused by the old client being closed, the CA watcher should just continue and use the new client rather than emitting an error that is confusing to users.

### CODE REVIEW

**Review by ravicious — APPROVED** [review]

File: lib/tbot/ca_rotation.go [hunk 1]
```diff
@@ -114,6 +117,18 @@ func (b *Bot) caRotationLoop(ctx context.Context) error {
...
+		// as part of the credentials renewal. Ignore it, and immediately begin
+		// watching again with the new client. We can safely check for Canceled
+		// here, because if the context was actually canceled, it would've
+		// been caught in the error check immediately following watchCARotations
+		var statusErr interface {
+			GRPCStatus() *status.Status
+		}
+		if errors.As(err, &statusErr) && statusErr.GRPCStatus().Code() == codes.Canceled {
+			continue
+		}
```
Comment (espadolini) [inline lib/tbot/ca_rotation.go:130]:
```suggestion
		if statusErr, _ := status.FromError(err); statusErr.Code() == codes.Canceled {
			continue
		}
```

Comment (espadolini) [reply lib/tbot/ca_rotation.go:130]: Also, are you sure we shouldn't keep the jitter at least?

**Review by espadolini — APPROVED** [review]

File: lib/tbot/ca_rotation.go (hunk 1 above)
Comment (strideynet) [reply lib/tbot/ca_rotation.go:130]: Agree on the jitter, but wouldn't your suggested change not work with wrapped errors ?

Comment (strideynet) [reply lib/tbot/ca_rotation.go]: Addressed by https://github.com/gravitational/teleport/pull/19266/commits/77ff82c5ba76a74205d30b8c9db9f491b03d9774

**Pull request merged**